### DISCUSSION TITLE
After using useNavigate to achieve page jump, how to destroy the previous page components

### DISCUSSION THREAD

**Discussion by lhs1478** [remix-run/react-router#9433] [Q&A]
I defined a timer in the main page, but when I use useNavigate to make the page jump to the login page, I found on the login page that the timer of the main page is still running.

The version I am using is 6.4.1.

**Comment by ryanflorence** [comment]
can `useLocation` to detect a page change and use that in a `useEffect` to cancel the timer, but w/o seeing the code not much else I can offer!
